# Project 2:  Lucid Titanic Sleuthing

## Part 1: Developing an understanding of the data

#### Based on the description of the data you read in the readme describe in your own words this data.

This data is going to contain the information of the people that were on the Titanic and Lusitania. We will be looking at the survival rate of the two data sets and see if there is any correlation between the two.

#### Based on our conceptual understanding of the columns in this data set.  What are the reasonable range of the values for the Sex, Age, SibSp, Parch columns.

In [1]:
#sibsp =sibling spouse, parch = parents and children

### Open the data in sublime text is there anything that jumps out to you?

## Part 2: reading the data in

#### Now read the data into a Pandas DataFrame

In [2]:
import pandas as pd
titanic='/Users/russellsasaki/Desktop/DSI/dsi_workspace/projects/project-02/assets/titanic.csv'
titanic_df=pd.read_csv(titanic)
titanic_df.head()

pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

     age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.00      0      0   24160  211.3375       B5        S    2    NaN   
1   0.92      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.00      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.00      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.00      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [11]:
lusitania='/Users/russellsasaki/Desktop/DSI/dsi_workspace/projects/project-02/assets/lusitania.csv'
lusitania_df=pd.read_csv(lusitania)
lusitania_df

Family name       Title       Personal name   Fate   Age  \
0        CAMERON         Mr.          Charles W.   Lost    38   
1     CARR-JONES         Mr.                  E.   Lost    37   
2      DRAKEFORD         Mr.              Edward  Saved    30   
3        HAWKINS         Mr.              Handel  Saved    25   
4      HEMINGWAY         Mr.        John William  Saved    27   
5       ANDERSON         Mr.        James Clarke   Lost    48   
6       ANDERSON         Mr.                John   Lost   NaN   
7         BATTLE         Mr.               James  Saved   NaN   
8         BESTIC         Mr.       Albert Arthur  Saved    24   
9         BOWDEN         Mr.             Joseph   Saved  19 ?   
10         BROWN         Mr.  John Manson George   Lost    57   
11        CANNON         Mr.        Robert Henry   Lost    50   
12       CLINTON         Mr.              George  Saved   NaN   
13         CLYDE         Mr.                Neil  Saved   NaN   
14         CRANK         Mr.                John   Lost   NaN   
15    CUNNINGHAM         Mr.              Edward  Saved   NaN   
16        DAVIES         Mr.                John  Saved   NaN   
17          EGAN         Mr.             William  Saved   NaN   
18        ELLIOT         Mr.       Arthur Graham   Lost   NaN   
19        ELLIOT         Mr.                Fred  Saved   NaN   
20         EVANS         Mr.      Thomas William   Lost   NaN   
21         FAYLE         Mr.              Edward  Saved   NaN   
22      FLEMMING         Mr.             Herbert   Lost   NaN   
23         GARRY         Dr.              Joseph   Lost    25   
24          GUNN         Mr.                  C.  Saved   NaN   
25        HARMAN         Mr.                John  Saved    56   
26       HEFFORD         Mr.               Percy   Lost   NaN   
27      HEIGHWAY         Mr.         Edward John  Saved    41   
28      HENNESEY         Mr.               Frank  Saved   NaN   
29        HERMAN         Mr.              Albert   Lost   NaN   
...          ...         ...                 ...    ...   ...   
1931      THOMAS        Mr.     William Stanford   Lost    44   
1932      THOMAS        Mr.         George Henry   Lost   NaN   
1933      THOMAS        Mr.                 John  Saved   NaN   
1934       THORN        Mr.       Alfred Richard   Lost   NaN   
1935    THORNLEY        Mr.               George  Saved   NaN   
1936       ULGAR        Mr.                 Jean   Lost   NaN   
1937   Van HOECK        Mr.           Julius E.   Saved    27   
1938      VARNEY        Mr.                 John   Lost   NaN   
1939      VERDIN        Mr.              Francis   Lost   NaN   
1940     WALLACE        Mr.                James  Saved   NaN   
1941       WEIGH        Mr.              Edward�  Saved   NaN   
1942        WEIR        Mrs.            Margaret   Lost    51   
1943     WHALLEY        Mr.               Leslie  Saved   NaN   
1944     WIGGINS        Mr.              William  Saved    23   
1945   WILKINSON        Mr.            Mr. James   Lost   NaN   
1946    WILLIAMS        Mr.                 John   Lost   NaN   
1947    WILLIAMS        Mr.           Mr. Harold   Lost   NaN   
1948    WILLIAMS        Mr.               Robert   Lost   NaN   
1949    WILLIAMS        Mr.       Robert Neptune   Lost   NaN   
1950    WILLIAMS        Mr.           William H.  Saved   NaN   
1951      WILMIN        Mr.              William  Saved   NaN   
1952      WILSON        Mr.               Robert   Lost   NaN   
1953     WISEMAN        Mr.                James   Lost   NaN   
1954        WOOD        Mr.               Walter  Saved    31   
1955        WOOD        Mr.         Henry Edward   Lost    35   
1956        WOOD        Mr.               Alfred  Saved   NaN   
1957        WOOD        Mr.        Wallace Edkin   Lost   NaN   
1958      WRIGHT  Master (?)       Harold Joseph   Lost   NaN   
1959       WYNNE      Master              George  Saved    16   
1960       WYNNE        Mr.       Joseph

#### Check that the age column doesn't have any unreasonable values 

In [10]:
titanic_df['age'].describe()       #1309 total passengers


count    1046.000000
mean       29.881138
std        14.413493
min         0.170000
25%              NaN
50%              NaN
75%              NaN
max        80.000000
Name: age, dtype: float64

For the Titanic data set, there are 263 missing ages from the data. With the pieces of data that were missing, they were reporting back a value of 'NAN'. When reporting back a value of 'NAN', it made it so the data set could not describe the upper percentiles of the data frame.

In [6]:
lusitania_df['Age'].describe()

count     1308
unique     120
top         30
freq        62
Name: Age, dtype: object

For the Lusitania data set, there are only 1308 ages counted out of 1961 total people that were in the boat. Of those counted, there are 120 unique values for the ages recorded. The unique values include the words 'months' and a '?'. Many of the missing ages in the data set are people that were part of the crew.

#### Check for missing values.  How do you know that a value is missing?

In [32]:
titanic_df.describe()

pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881138     0.498854     0.385027   
std       0.837836     0.486055    14.413493     1.041658     0.865560   
min       1.000000     0.000000     0.170000     0.000000     0.000000   
25%       2.000000     0.000000          NaN     0.000000     0.000000   
50%       3.000000     0.000000          NaN     0.000000     0.000000   
75%       3.000000     1.000000          NaN     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%            NaN         NaN  
50%            NaN         NaN  
75%            NaN         NaN  
max     512.329200  328.000000

The missing values are annotated as 'NAN' in the Titanic data set. When running the data set with the 'NAN' values, it makes it so that data can not count upper percentiles. 

In [12]:
lusitania_df.describe()

Value
count  1961.000000
mean      0.999490
std       0.022582
min       0.000000
25%       1.000000
50%       1.000000
75%       1.000000
max       1.000000

The missing values in the Lusitania data are also annotated as NAN. The values missing when describing the data are counted as a zero value for this data set, which make the describe portion of the data use those ages as a value of '0'. This makes it count the data at come with a mean, standard deviation and other values not show the true averages of  the ages of the people in this data set.

#### Does it makes sense to guess at the value?

For much of the Titanic data, it is not a good idea to make a guess at the 'NAN' value. There is 20% of the 'age' data that is missing. Giving every person the average age of 30 does not work well for 20% of the missing data. 

For the Lusitania data, using the mean value for age is not going to work because the values of '0' affect the mean too much.

## Part 3: data imputation

#### Well let’s say that it does... You likely noticed that Age has some missing values. How many are missing?

Titanic missing ages: 263
Luistania missing ages: 653

#### For the Age of the passangers ... how would you guess at the missing values using the other data present in the CSV.

In [ ]:
#.fillNA

## Part 4: Group Statistics

#### Are there any groups that were especially adversely affected in the Titanic wreck? (justify your response numerically)

#### Are there any groups that outperformed the survival of the latter group? (justify your response numerically)


## Part 5:  Comparative Statistics:  Lusitania

#### Are there any groups that were especially adversely affected in the Titanic wreck? (justify your response numerically)

#### Are there any groups that outperformed the survival of the latter group? (justify your response numerically)



#### What does the group-wise survival rate imply about circumstances during these two accidents?